wren-ai-service/eval/data_curation/utils.py

```diff
@@ -0,0 +1,426 @@
+                    columns_ddl.append(fk_constraint)
+
+            if "properties" in model:
+                comment = (
+                    f"\n/* {orjson.dumps(model['properties']).decode("utf-8")} */\n"
+                )
+            else:
+                comment = ""
+
+            create_table_ddl = (
+                f"{comment}CREATE TABLE {table_name} (\n  "
+                + ",\n  ".join(columns_ddl)
+                + "\n);"
+            )
+            ddl_commands.append(create_table_ddl)
+
+        return ddl_commands
+
+    def _convert_metrics(metrics: List[dict]):
+        ddl_commands = []
+
+        for metric in metrics:
+            table_name = metric["name"]
+            columns_ddl = []
+            for dimension in metric["dimension"]:
+                column_name = dimension["name"]
+                column_type = dimension["type"]
+                comment = "-- This column is a dimension\n  "
+                column_ddl = f"{comment}{column_name} {column_type}"
+                columns_ddl.append(column_ddl)
+
+            for measure in metric["measure"]:
+                column_name = measure["name"]
+                column_type = measure["type"]
+                comment = f"-- This column is a measure\n  -- expression: {measure["expression"]}\n  "
+                column_ddl = f"{comment}{column_name} {column_type}"
+                columns_ddl.append(column_ddl)
+
+            comment = f"\n/* This table is a metric */\n/* Metric Base Object: {metric["baseObject"]} */\n"
+            create_table_ddl = (
+                f"{comment}CREATE TABLE {table_name} (\n  "
+                + ",\n  ".join(columns_ddl)
+                + "\n);"
+            )
+
+            ddl_commands.append(create_table_ddl)
+
+        return ddl_commands
+
+    def _convert_views(views: List[dict]):
+        def _format(view: dict[str, Any]) -> str:
+            properties = view["properties"] if "properties" in view else ""
+            return f"/* {properties} */\nCREATE VIEW {view['name']}\nAS ({view['statement']})"
+
+        return [_format(view) for view in views]
+
+    models = mdl_json.get("models", [])
+    relationships = mdl_json.get("relationships", [])
+    metrics = mdl_json.get("metrics", [])
+    views = mdl_json.get("views", [])
+
+    ddl_commands = (
+        _convert_models_and_relationships(models, relationships)
+        + _convert_metrics(metrics)
+        + _convert_views(views)
+    )
+    return "\n\n".join(ddl_commands)
+
+
+async def get_sql_analysis(
+    sql: str,
+) -> List[dict]:
+    sql = sql[:-1] if sql.endswith(";") else sql
+    async with aiohttp.request(
+        "GET",
+        f"{WREN_ENGINE_ENDPOINT}/v1/analysis/sql",
+        json={
+            "sql": add_quotes(sql),
+            "manifest": st.session_state["mdl_json"],
+        },
+        timeout=aiohttp.ClientTimeout(total=TIMEOUT_SECONDS),
+    ) as response:
+        return await response.json()
+
+
+async def get_contexts_from_sqls(
```

Comment:
good to me this function. I think I could using in the metrics as well. really appreciate your work! 👍